stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MediaAttachmentView.kt

```diff
@@ -21,12 +26,17 @@ import io.getstream.chat.android.ui.common.extensions.internal.streamThemeInflat
 import io.getstream.chat.android.ui.common.style.setTextStyle
 import io.getstream.chat.android.ui.databinding.StreamUiMediaAttachmentViewBinding
 import io.getstream.chat.android.ui.message.list.adapter.view.MediaAttachmentViewStyle
+import kotlin.math.min
 
 internal class MediaAttachmentView : ConstraintLayout {
     var attachmentClickListener: AttachmentClickListener? = null
     var attachmentLongClickListener: AttachmentLongClickListener? = null
     var giphyBadgeEnabled: Boolean = true
 
+    private val maxMediaAttachmentWidth: Int by lazy {
+        (Resources.getSystem().displayMetrics.widthPixels * MAX_WIDTH_PERCENTAGE).toInt()
```

Comment:
`Resources.getSystem().displayMetrics.widthPixels` is the width of the entire screen, right? Should (could) we take the width of the current `View` instead? For example, what happens if there's something else next to it on one of the sides?

Reply:
At this moment the view is not drawn yet, so the width and height are 0 =|, but we should use `messageMaxWidthFactorMine` and `messageMaxWidthFactorTheirs`, I forgot to update this to use those. 

About view next to MessageListView... That's a complicated one, because we don't know the size of the view before it is actually drawn. Maybe a configuration to set a max possible width?

Reply:
Are you sure the width of the View wouldn't be available by the time we need to calculate this value? In that case, perhaps we should wait until the View is layed out, and only then run any code that lays out the Gif contained within?

Reply:
It just feels incorrect to base this sizing on the display size, as that's not what we actually want to do - we want to size based on the containing View's size.